### PR TITLE
[New Segment] - add prompt segment for nodenv

### PR DESCRIPTION
**Pull request by greg-cook**
#### Description
- Adds a prompt segment for [`nodenv`](https://github.com/nodenv/nodenv)
- Copied from existing [`rbenv`](https://github.com/bhilburn/powerlevel9k/blob/next/segments/rbenv.p9k) segment

#### Questions
Please let me know whether specs are sufficient, copied from [nvm segment spec](https://github.com/bhilburn/powerlevel9k/blob/next/test/segments/nvm.spec) with some removed due to differences in the way they work.

#### Screenshot
image

### CODE REVIEW

**Comment by dritter** [comment]
Hi @greg-cook !
Thanks for your contribution. One question: Is `nodenv` related to `nodeenv` (which we already have a segment for).. The current segment refers to [this project](https://github.com/ekalinin/nodeenv).

File: segments/nodenv.p9k [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# @description
+#   Display the current nodenv information.
+##
+# @args
+#   $1 string Alignment - left | right
+#   $2 integer Segment index
+#   $3 boolean Whether the segment should be joined
+##
+prompt_nodenv() {
+  if [[ -n "${NODENV_VERSION}" ]]; then
```
Comment (dritter) [inline segments/nodenv.p9k]: Despite looking very convenient, that shortcut ignores `$P9K_NODENV_PROMPT_ALWAYS_SHOW`. So, if somebody uses that setting, it won't work once `NODENV_VERSION` is set (regardless, if this is the same version as `nodenv global`).

File: segments/nodenv.p9k [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    p9k::prepare_segment "$0" "" $1 "$2" $3 "${NODENV_VERSION}"
+  elif [ ${commands[nodenv]} ]; then
+    local nodenv_version_name="$(nodenv version-name)"
```
Comment (dritter) [inline segments/nodenv.p9k]: Not sure how fast `nodenv` is, but we call it three times here. Is there a way to get the infos at once? Or a way to get the infos in a different way (thinking of our `__p9k_upsearch ` function)?

**Comment by romkatv** [comment]
@greg-cook Would you mind sharing why you closed this PR?

**Comment by greg-cook** [comment]
Honestly, I haven't been able to spend any time on this (and at this point I'm no longer using `nodenv`)

It is of course still a feature worth having if somebody else wanted to pick it up!

**Comment by Larusso** [comment]
Funny I just wanted to open a patch for this as well.
I think it would make sense to support `nodenv` along with `rbenv` and `pyenv`.
I would even add `jenv` for that matter. So could we integrate it?

**Comment by romkatv** [comment]
It's a remarkable coincidence that everyone who requests integration with `nodenv` has cookie monster for avatar. Or is it? :thinking:

**Comment by kyledakid** [comment]
I need the nodenv segment also :( I dont want to use nodeenv (double e), it relies on virtualenv.
@greg-cook can you pls share me a way for this ?

**Pull request closed without merging**